tsdb/db.go

```diff
@@ -732,6 +732,10 @@ func (db *DB) run() {
 
 		select {
 		case <-time.After(1 * time.Minute):
+			if err := db.reloadBlocks(); err != nil {
```

Comment:
We should be protecting it with `db.cmtx` lock here.

Reply:
I've added a commit with the lock :)

But what exactly does this particular lock do? It does not allow compactions to happen while the mutex is locked?

Reply:
> It does not allow compactions to happen while the mutex is locked?

Yes. `reloadBlocks` indirectly depends on the compaction mutex to avoid races since we did reload only during compaction until now.